alvr/client_openxr/src/interaction.rs

```diff
@@ -43,223 +65,222 @@ pub struct BodySources {
     pub body_tracker_fb: Option<(BodyTrackerFB, usize)>,
 }
 
+#[derive(Clone)]
+pub struct InteractionSourcesConfig {
+    pub face_tracking: Option<FaceTrackingSourcesConfig>,
+    pub body_tracking: Option<BodyTrackingSourcesConfig>,
+    pub prefers_multimodal_input: bool,
+}
+
+impl InteractionSourcesConfig {
+    pub fn new(config: &StreamConfig) -> Self {
+        Self {
+            face_tracking: config
+                .settings
+                .headset
+                .face_tracking
+                .as_option()
+                .map(|c| c.sources.clone()),
+            body_tracking: config
+                .settings
+                .headset
+                .body_tracking
+                .as_option()
+                .map(|c| c.sources.clone()),
+            prefers_multimodal_input: config
+                .settings
+                .headset
+                .controllers
+                .as_option()
+                .map(|c| c.multimodal_tracking)
+                .unwrap_or(false),
+        }
+    }
+}
+
 pub struct InteractionContext {
+    xr_session: xr::Session<xr::OpenGlEs>,
+    platform: Platform,
     pub action_set: xr::ActionSet,
     pub button_actions: HashMap<u64, ButtonAction>,
     pub hands_interaction: [HandInteraction; 2],
-    pub uses_multimodal_hands: bool,
+    pub multimodal_hands_handle: Option<MultimodalHandsHandle>,
```

Comment:
This would need to be changed when doing my suggestion to an extra function that actually checks whether it's active